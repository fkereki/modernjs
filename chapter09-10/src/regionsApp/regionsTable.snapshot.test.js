/* @flow */

import React from "react";
import TestRenderer from "react-test-renderer";

import { RegionsTable } from "./regionsTable.component";

describe("RegionsTable", () => {
    it("renders correctly an empty list", () => {
        const tree = TestRenderer.create(
            <RegionsTable list={[]} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders correctly a list", () => {
        const tree = TestRenderer.create(
            <RegionsTable
                list={[
                    {
                        countryCode: "UY",
                        regionCode: "10",
                        regionName: "Montevideo"
                    },
                    {
                        countryCode: "UY",
                        regionCode: "9",
                        regionName: "Maldonado"
                    },
                    {
                        countryCode: "UY",
                        regionCode: "5",
                        regionName: "Cerro Largo"
                    }
                ]}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
