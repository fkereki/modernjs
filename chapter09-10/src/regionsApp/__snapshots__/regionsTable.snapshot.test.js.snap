// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegionsTable renders correctly a list 1`] = `
<div
  className="bordered"
>
  <div>
    Cerro Largo
  </div>
  <div>
    Maldonado
  </div>
  <div>
    Montevideo
  </div>
</div>
`;

exports[`RegionsTable renders correctly an empty list 1`] = `
<div
  className="bordered"
>
  No regions.
</div>
`;
